src/passes/inlineAssemblyTransformer.ts

```diff
@@ -0,0 +1,124 @@
+import { ASTMapper } from '../ast/mapper';
+import { AST } from '../ast/ast';
+import {
+  createExpressionStatement,
+  createIdentifier,
+  createNumberLiteral,
+  createTuple,
+} from '../utils/nodeTemplates';
+
+import {
+  InlineAssembly,
+  YulNode,
+  Identifier,
+  ASTNode,
+  BinaryOperation,
+  Assignment,
+  Block,
+  Expression,
+  ExpressionStatement,
+  VariableDeclaration,
+  UncheckedBlock,
+  Literal,
+  Statement,
+} from 'solc-typed-ast';
+import assert from 'assert';
+import { WillNotSupportError } from '../export';
+
+class YulTransformer {
+  vars: Map<string, VariableDeclaration>;
+  assemblyRoot: InlineAssembly;
+  ast: AST;
+
+  constructor(ast: AST, assemblyRoot: InlineAssembly) {
+    this.ast = ast;
+    this.assemblyRoot = assemblyRoot;
+    const blockRefIds = assemblyRoot.externalReferences.map((ref) => ref.declaration);
+    const blockRefs = [...assemblyRoot.context!.map].filter(([id, _]) =>
+      blockRefIds.includes(id),
+    ) as [number, VariableDeclaration][];
+    this.vars = new Map(blockRefs.map(([id, ref]) => [ref.name, ref]));
+  }
+
+  run(node: YulNode): Statement {
+    const methodName = `create${node.nodeType}`;
+    if (!(methodName in this))
+      throw new WillNotSupportError(`${node.nodeType} is not supported`, this.assemblyRoot, false);
+    const method = this[methodName as keyof YulTransformer] as (node: YulNode) => ASTNode;
```

Comment:
It seems to be inconsistent with [https://github.com/NethermindEth/warp/blob/develop/src/ast/visitor.ts#L87](https://github.com/NethermindEth/warp/blob/develop/src/ast/visitor.ts#L87) as pointed out by @rodrigo-pino. I prefer naming methods as `node.nodeType` (without the create prefix) and map them as it is implemented above

Reply:
I think we should keep the create prefix, because its a verb instead of just the nodeType which is a noun.
The verb makes its obvious what the function will do

Reply:
I prefer your current approach rather than the one we use in `ASTVisitor` with all of the `if`s. I do think that the whole project should be consistent, so I would say to change the approach on `ASTvisitor` for this one (I don't see atm any reason why we couldn't do so), but that should be left to do in a different PR.

About the way you do the mapping, I like to use a prefix as well, but I would use something like `transform` instead of `create`. First, because it would then be similar to the other methods like `transformNode`. Second, because you are not actually "creating" a yulNode but "transforming" it into an ASTNode